pkg/codegen/docs/gen_creation_example_for_yaml.go

```diff
@@ -0,0 +1,193 @@
+		}
+		if mod := strings.Split(tokenParts[1], "/"); len(mod) == 2 && title(mod[1]) == tokenParts[2] {
+			// aws:s3/bucket:Bucket => aws:s3:Bucket
+			// We recourse to handle the case foo:index/bar:Bar => foo:index:Bar
+			tokenParts = []string{tokenParts[0], mod[0], tokenParts[2]}
+		}
+
+		if tokenParts[1] == "index" || tokenParts[1] == "" {
+			// foo:index:Bar => foo:Bar
+			// or
+			// foo::Bar => foo:Bar
+			tokenParts = []string{tokenParts[0], tokenParts[2]}
+		}
+	}
+
+	return strings.Join(tokenParts, ":")
+}
+
+func genCreationExampleSyntaxYAML(r *schema.Resource) string {
+	indentSize := 0
+	buffer := bytes.Buffer{}
+	write := func(format string, args ...interface{}) {
+		buffer.WriteString(fmt.Sprintf(format, args...))
+	}
+
+	indent := func() {
+		buffer.WriteString(strings.Repeat(" ", indentSize))
+	}
+
+	indended := func(f func()) {
+		indentSize += 2
+		f()
+		indentSize -= 2
+	}
+
+	seenTypes := codegen.NewStringSet()
+	var writeValue func(valueType schema.Type)
+	writeValue = func(valueType schema.Type) {
+		switch valueType {
+		case schema.BoolType:
+			write("True|False")
+		case schema.IntType:
+			write("0")
+		case schema.NumberType:
+			write("0.0")
+		case schema.StringType:
+			write("\"string\"")
+		case schema.ArchiveType:
+			write("\n")
+			indended(func() {
+				indent()
+				write("Fn::FileAsset: ./file.txt")
+			})
+		case schema.AssetType:
+			write("\n")
+			indended(func() {
+				indent()
+				write("Fn::StringAsset: \"example content\"")
+			})
+		}
+
+		switch valueType := valueType.(type) {
+		case *schema.ArrayType:
+			if isPrimitiveType(valueType.ElementType) {
+				write("[")
+				writeValue(valueType.ElementType)
+				write("]")
+			} else {
+				write("[")
+				writeValue(valueType.ElementType)
+				write("\n")
+				indent()
+				write("]")
+			}
+		case *schema.MapType:
+			write("\n")
+			indended(func() {
+				indent()
+				write("\"string\": ")
+				writeValue(valueType.ElementType)
+				if !isPrimitiveType(valueType.ElementType) {
+					write("\n")
+				}
+			})
+		case *schema.ObjectType:
+			if seenTypes.Has(valueType.Token) && objectTypeHasRecursiveReference(valueType) {
+				write("type(%s)", valueType.Token)
+				return
+			}
+
+			seenTypes.Add(valueType.Token)
+			write("\n")
+			indended(func() {
+				for index, p := range valueType.Properties {
+					indent()
+					write("%s: ", p.Name)
+					writeValue(p.Type)
+					if index != len(valueType.Properties)-1 {
+						write("\n")
+					}
+				}
+			})
+		case *schema.ResourceType:
+			write("reference(%s)", valueType.Token)
+		case *schema.EnumType:
+			cases := make([]string, len(valueType.Elements))
+			for index, c := range valueType.Elements {
+				if stringCase, ok := c.Value.(string); ok && stringCase != "" {
+					cases[index] = stringCase
+				} else if intCase, ok := c.Value.(int); ok {
+					cases[index] = strconv.Itoa(intCase)
+				} else {
+					if c.Name != "" {
+						cases[index] = c.Name
+					}
+				}
+			}
+
```

Comment:
The example in the decscripton seems to be illegal code becuase the string values aren't quoted (and the variables aren't bound to anything)?

```
LogType: AuditLogConfigLogTypeLogTypeUnspecified|AuditLogConfigLogTypeAdminRead|AuditLogConfigLogTypeDataWrite|AuditLogConfigLogTypeDataRead,

```

Reply:
It is true that the the enum here should be qualified by package and module if necessary. However, I am wondering if, at least for go, we can a single enum case in the code, then add a comment to show the other cases of the enum:
```go
import (
  "github.com/pulumi/pulumi/sdk/v3/go/pulumi"
  "github.com/pulumi/pulumi-example/sdk/v3/go/example/myModule"
)

iamResource, err := myModule.NewIamResource("iamResource", &myModule.IamResourceArgs{
  Config: &iam.v1.AuditConfigArgs{
    AuditLogConfigs: iam.v1.AuditLogConfigArray{
      &iam.v1.AuditLogConfigArgs{
        ExemptedMembers: pulumi.StringArray{
          pulumi.String("string")
        },
        // can be AdminRead|DataWrite|DataRead
        LogType: exammple.myModule.AuditLogConfigLogTypeLogTypeUnspecified
      }
    },
    Service: pulumi.String("string"),
  },
})
```
what do you think? the comment can show the entire fully qualified enum or just the case name.

Reply:
I like the comment much more than the full version.